CIP-0094/README.md

```diff
@@ -0,0 +1,281 @@
+
+```cbor
+question =
+  { 0: prompt
+  , 1: [ * choice ]
+  , ? "_": nonce
+  }
+
+prompt =
+  [ * text .size (0..64) ]
+
+choice =
+  [ * text .size (0..64) ]
+
+nonce =
+  uint
+```
+
+A nonce is optionally included to provide non-replayability should the same question and answers be asked multiple times over different periods. The transaction carrying a question **must** be signed by one of the genesis delegate keys to be considered valid. This genesis key signature isn't captured in the metadata but in the transaction itself as an extra signatory.
+
+For example:
+
+<table>
+<thead>
+  <th>CBOR diagnostic</td>
+  <th>Base16-encoded</th>
+</thead>
+<tbody>
+ <tr>
+  <td>
+<pre>
+{ 94:
+  { 0: [ "Pineapples on pizza?" ]
+  , 1:
+    [ [ "yes" ]
+    , [ "no" ]
+    ]
+  }
+}
+</pre>
+  </td>
+  <td>
+   <a target="_blank" href="https://cbor.me/?bytes=A1185EA200817450696E656170706C6573206F6E2070697A7A613F0182816379657381626E6F">
+<pre>
+A1185EA200817450696E656170706C6573206F
+6E2070697A7A613F0182816379657381626E6F
+</pre>
+   </a>
+  </td>
+ </tr>
+</tbody>
+</table>
+
+### Answer structure
+
+Similarly, an answer to a question is posted as transaction's metadata using the label `94` and the following metadata structure:
+
+```cbor
+answer =
+  { 2: question_hash
+  , 3: choice
+  }
+
+question_hash =
+  bytes .size 32
+```
+
+Some remarks:
+
+1. The field `2` (`question_hash`) is a blake2b-256 hash digest, whose preimage is the entire serialised question metadata payload (with the `94` top-level label).
+1. The field `3` represents the 0-based index of the chosen answer from the available choices (from field `1` of the target poll).
+
+For example:
+
+<table>
+<thead>
+  <th>CBOR diagnostic</td>
+  <th>Base16</th>
+</thead>
+<tbody>
+ <tr>
+  <td>
+<pre>
+{
+  94: {
+    2: h'29093fd43fc30ba31e306af06ce8537390e1668ae7496fe53d53684683c3762c',
+    3: 0
+  }
+}
+</pre>
+  </td>
+  <td>
+   <a target="_blank" href="https://cbor.me/?bytes=A1185EA202582029093FD43FC30BA31E306AF06CE8537390E1668AE7496FE53D53684683C3762C0300">
+<pre>
+A1185EA202582029093FD43FC30BA31E306AF06CE
+8537390E1668AE7496FE53D53684683C3762C0300
+</pre>
+   </a>
+  </td>
+ </tr>
+</tbody>
+</table>
+
+The transaction carrying the answer metadata must then **be signed using a stake pool operator cold key**. Because cold key are not payment keys, this requires specifying an extra required signer on the transaction (transaction's field number 14 as per [Babbage's CDDL](https://github.com/input-output-hk/cardano-ledger/blob/cffa75fdbd800cda60997791e51bf02f2af0c42b/eras/babbage/test-suite/cddl-files/babbage.cddl#L66)). The signature shall be produced in an air-gapped environment only.
+
+> **Warning**
+>
+> Only the first answer to a poll for each credential shall be considered. If multiple answers are found, only the first answer submitted (transaction & block ordering tallying) shall be considered.
+
+### Procedure and Duration
```

Comment:
For me, what is missing from this specification is an explanation of the full poll flow, for example: 

- how can SPOs can find new polls to vote on?
- how any ecosystem actor can verify the results of a poll?
- where does the report get posted?

Reply:
We've addressed those points in separate documents and tutorial by now. Thing is, I don't necessarily want to take a strong stance about these in the CIP itself to not force one single approach. 

The data lives on chain as transaction metadata which means that it's just the usual approach for any tool or builder to get access to it. We now even have dashboards in at least two explorers (adastat and cardanoscan) that are collecting and validating these data. And it's relatively straightforward with a chain sync. More tools will likely follow.

Reply:
At the moment, every question with the label 94 appears on-chain for all to see. Technically, from this moment on, anyone - with a pool s key - can pick up this question and submit an answer. It is up to the publishing entity to set the cast and redelegation duration. But it is not up to this one entity to interpret or publish the result, because it has no right to decide about the interpretation or reaction of the other entities. 

Like the question, all individual ballots are transparent and verifiable on-chain. The transaction format with the `--required-signer-hash` ensures that only ballots validly signed by a pool end up on chain. 

To make the on-chain data easily accessible for those who do not have a full indexing instance (e.g. dbsync) available, the CardanoFoundation has set up a github repository where an index file in json format is maintained. There you can find (for now) the cast/delegation epochs in a standardized machine-readable format. In addition, there is also the possibility to store list of all ballots that were considered valid summarized in one file, in order to give everyone easy access and the possibility of their own analysis and interpretation.

The methodology deliberately chosen in this poll design is a creative+inclusive+motivational iteration. This perhaps somewhat contradicts the common expectation that a CIP must be written (first define, then implement, then execute) 
In this process, we deliberately wanted to present a maximally simple, straight executable thing, with lots of room for ideas. People should have the opportunity to get hands-on experience during the build process and participate in the rest of the design. At the beginning of the project it was still unclear whether this was only a one-time test run with the goal of incorporating the data and findings into the design of CIP-1694, or whether it would make sense to refine this CIP step by step for further test runs.